asllib/Typing.ml

```diff
@@ -2217,18 +2336,24 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
                 (Error.BadArity
                    (Static, "LEDestructuring", List.length tys, List.length les))
             else
-              let les' = List.map2 (annotate_lexpr env) les tys in
-              LE_Destructuring les' |> here
+              let les', sess =
+                List.map2 (annotate_lexpr env) les tys |> List.split
+              in
+              let ses =
+                (* TODO left-hand-side conflicting union *)
```

Comment:
I don't understand the comment. What's precenting you from using conflicting union here?